Replace slider with sat amount buttons when sending

secondl1ght opened the pull request:
As requested in #529, I have removed the slider on the sending sats screen. I replaced it with 4 buttons with pre-determined satoshi amounts for each: 10, 100, 1000 and 10000. For the functionality I made it so that when you click each button it adds the satoshi amount to the input, I thought this may be useful rather than only allowing for the number to be set once.

What does everyone think about this?

Screenshot:
![image](https://user-images.githubusercontent.com/85003930/150292621-9d7bb8dd-1eae-4769-b202-6bdec113bd1b.png)

I think there is another screen I will have to adjust but I thought I would get feedback on this first and if everyone likes it then I can update the other one as well.

dylancom commented:
<img width="344" alt="Schermafbeelding 2022-01-20 om 09 05 19" src="https://user-images.githubusercontent.com/12894112/150297923-42d2994f-3a64-4c5e-ac92-8f5651688dc8.png">
Some inspiration from bolt.fun

bumi commented:
that's actually great. I like it!! 
best to keep it in one line probably

bumi commented on src/app/screens/LNURLPay.tsx:
```diff
@@ -47,7 +47,7 @@ function LNURLPay(props: Props) {
     LNURLPaymentSuccessAction | undefined
   >();
-
+  const [clear, setClear] = useState<boolean>(false);
```
is this needed? 
if we make sure that `valueMSat` is never undefined but always a integer then we
can simple always do `setValueMSat(valueMSat + 10000);` I guess?

secondl1ght replied:
The reason I added this was because currently `valueMSat` starts off with a value of the minimum sendable. So for me when I was testing there was already a value of 1 in the input. This made it so that when I clicked the buttons instead of going up in even numbers like 10 or 1000 it would do 11, 1001 etc. So I thought it would be cleaner to clear what is the value at the start so that when you use the buttons it reacts the way you would expect. Does this make sense?

secondl1ght commented:
What do you guys think of this?

secondl1ght commented:
I was also thinking we could make these buttons light up on click/active to a lightning yellow color so when you click them it looks like there is lightning striking? ⚡⚡

dylancom commented:
I agree with Bumi to keep it on 1 line. No need for 2 rows.

secondl1ght commented:
Ok sounds good, I didn't think I'd be able to fit everything on 1 row but it looks like we can:
![image](https://user-images.githubusercontent.com/85003930/150696610-62cd2bbf-7f2d-4f9d-98cf-aa7ddb97e0cc.png)

dylancom commented:
Then I would remove the gap on the  left of the first button and the right of the last button so it aligns nicely with the amount field (just like the bolt.fun example)

secondl1ght commented:
How's this?
![image](https://user-images.githubusercontent.com/85003930/150697208-15fe3b8b-f5eb-4a39-ae62-08daf645aeb7.png)

dylancom commented:
Much better! Bonus points if the horizontal spacing between the buttons is the same as the vertical spacing between the amount field and the buttons (i'd say).

secondl1ght commented:
Woot!
![image](https://user-images.githubusercontent.com/85003930/150697516-433e500a-943f-49c3-b3c0-cefbf0bcedd8.png)

secondl1ght commented:
Do you know where the other slider is in the app? I see it in the code but I'm not sure how to navigate to it on Alby...

secondl1ght commented:
Latest screenshot:
![image](https://user-images.githubusercontent.com/85003930/150699372-12d38469-c0a3-4bf4-b5cf-afa03ad2239b.png)

secondl1ght commented:
Ok I have updated the functionality to only set the input one time, I also updated the values of the buttons.
![image](https://user-images.githubusercontent.com/85003930/150810406-cc2672ca-7380-4802-a50b-7bc2657a2461.png)

secondl1ght commented:
> instead of a lessPadding prop we could opt for a noPadding prop, and let flex stretch the buttons to available space.

Here's how it looks with no padding, I think it looks better with padding personally:
![image](https://user-images.githubusercontent.com/85003930/150812567-e1cc2f73-8138-490d-a5d0-d350ab9be271.png)

Or do I have something wrong with flex and it is not filling the width properly?

dylancom commented:
You should remove the extra div's around the buttons. Add the fullWidth props to the buttons and remove "flex-wrap" from the parent wrapper.

secondl1ght commented:
Oh right, I had those extra divs in there earlier to apply styling but wasn't using them anymore, good call. Here is how it looks now:
![image](https://user-images.githubusercontent.com/85003930/150818096-364c00dd-9780-4e0f-8400-685f94146223.png)

To add X margin should I create another prop on the Button component called marginX?

secondl1ght commented:
Yeah! I forgot about that class, I have to remember all these useful tricks.

secondl1ght commented:
Ok I have also updated the slider on the make invoice screen as well:
![image](https://user-images.githubusercontent.com/85003930/150828225-c2f6ce5f-1abe-48b6-8955-156899a58114.png)

For this one I did add the validation for the min/max because that is part of the invoice.

I think this PR is now ready for final review and then merging if all looks good.

dylancom commented:
<img width="371" alt="Schermafbeelding 2022-01-24 om 18 16 18" src="https://user-images.githubusercontent.com/12894112/150831809-6c80aa1d-fd66-4fe7-bc93-07dde8a3bc0f.png">

Tigerix commented:
Suggestion for the amount buttons:
Clicking the 100 Sats button 3 times in a row, should result in 300 Sats!
Clicking it 10 times, should result in 100 Sats again.
